src/applications/personalization/profile-2/components/ProfileInfoTable.jsx

```diff
@@ -51,35 +52,41 @@ const ProfileInfoTable = ({
   const tableRowDataClasses = ['vads-u-margin--0'];
 
   return (
-    <div className={tableClasses.join(' ')} data-field-name={fieldName}>
+    <section className={tableClasses.join(' ')} data-field-name={fieldName}>
       {title && (
         <h3 className={[...titleClasses, ...titleClassesMedium].join(' ')}>
           {title}
         </h3>
       )}
-      {data
-        .map(element => (dataTransformer ? dataTransformer(element) : element))
-        .map((row, index) => (
-          <div
-            key={index}
-            className={`'table-row' ${[
-              'table-row',
-              ...tableRowClasses,
-              ...tableRowClassesMedium,
-            ].join(' ')}`}
-          >
-            <h4
-              className={[
-                ...tableRowTitleClasses,
-                ...tableRowTitleMediumClasses,
-              ].join(' ')}
+      {/* This syntax is kind of ugly. But it's simply adding `role="list"` if `list` is truthy` */}
+      <dl className="vads-u-margin--0" {...{ ...(list && { role: 'list' }) }}>
+        {data
+          .map(
+            element => (dataTransformer ? dataTransformer(element) : element),
+          )
+          .map((row, index) => (
+            <div
+              key={index}
+              className={`table-row ${[
+                ...tableRowClasses,
+                ...tableRowClassesMedium,
+              ].join(' ')}`}
             >
-              {row.title}
-            </h4>
-            <div className={tableRowDataClasses.join(' ')}>{row.value}</div>
-          </div>
-        ))}
-    </div>
+              <dt
+                className={[
+                  ...tableRowTitleClasses,
+                  ...tableRowTitleMediumClasses,
+                ].join(' ')}
+                // again this is weird syntax but it's adding `role="listitem"` if the `list` prop is truthy
+                {...{ ...(list && { role: 'listitem' }) }}
```

Comment:
👀

Reply:
Since this `role` condition is used twice it could be cleaner to have it moved outside the JSX. It is also a little funky bc of the two spread operators

Reply:
Moved outside of the JSX how? I think that the alternative would be have all the logic for setting up this `dt` element live outside of the `return` block, but that also doesn't seem great? I'll take a stab at that and see how it looks.

Reply:
Yeah I was thinking as just a constant defined just outside the `return` statement, nothing fancy. And yeah, just preference, but isn't necessarily an improvement 😸